tests/test_visitors.py

```diff
@@ -205,3 +237,73 @@ def test_transformer_add_replace(exprs, block2, block3):
         assert line1 in newcode
         assert line2 in newcode
         assert "a[i0] = a[i0] + b[i0] + 5.0F;" not in newcode
+
+
+def test_merge_iterations_flat(exprs, iters):
+    """Test outer loop merging on a simple two-level hierarchy:
+
+    for i                       for i
+        for j              \        for j
+            expr0       === \           expr0
+    for i               === /       for k
+        for k              /            expr1
+            expr1
```

Comment:
@mlange05 Sorry, I explained myself really badly: what I meant was to leave a TODO comment in ``StencilKernel``, so that we remember to deal with this in the near future *after* this PR has been merged !